src/porepy/numerics/ad/operators.py

```diff
@@ -1864,16 +1876,24 @@ def __init__(self, variables: list[Variable]) -> None:
         # Default values for empty md variable
         else:
             time_indices = [-1]
-            iterate_indices = [0]
+            iter_indices = [-1]
             names = ["empty_md_variable"]
+            current_iter = [True]
 
         ### PRIVATE
         self._id = next(Variable._ids)
         # NOTE private time step index is -1 if public time step index of atomic
         # variables is None (current time)
         self._time_step_index = -1 if time_indices[0] is None else time_indices[0]
-        # NOTE private and public iterate indices are always integers
-        self._iterate_index = iterate_indices[0]
+
+        # If current time and iterate
+        if current_iter[0]:
+            self._iterate_index = -1  # current time and iter
```

Comment:
In this case, isn't iter_indices[0] None?

Reply:
If all sub variables are current iterate, then `iter_indices[0] == 0`, because they fetch their `.val` from that index during `.value_and_jacobian()`.

Reply:
Discussed in person, added documentation why this is required and the if-else branch cannot be removed.